pandas/tools/util.py

```diff
@@ -61,6 +61,42 @@ def to_numeric(arg, errors='raise'):
         - If 'raise', then invalid parsing will raise an exception
         - If 'coerce', then invalid parsing will be set as NaN
         - If 'ignore', then invalid parsing will return the input
+    downcast : {'signed', 'unsigned'}, str, numpy.dtype, or
+               Python dtype, default None
+        If possible, downcast the resulting numerical data to the
+        specified numerical dtype in 'downcast'. If the size
+        of the data's dtype is smaller or equal to that of 'downcast',
+        then this parameter will be ignored.
+
+        If 'downcast' refers to a more generic dtype (e.g. int),
+        downcast the resulting numerical data to the smallest
+        dtype possible under that more generic dtype specification
+        that is of smaller size than the current numerical
+        data dtype according to the following rules:
+
+        1) int
+
+           Smallest signed int dtype. To select this option,
+           pass in 'int', 'signed', np.int, or int.
+
+        2) uint
+
+           Smallest unsigned int dtype. To select this option,
+           pass in 'uint' or 'unsigned'.
+
+        3) float
+
+           Smallest float dtype. To select this option,
+           pass in 'float', np.float, or float.
+
+        If the specification cannot be followed (e.g. trying to
```

Comment:
@jreback : The `errors` argument is for the core casting of the raw data to numerical data (i.e. when the call is made to `lib.maybe_convert_numeric`).  Numeric conversion is coerced when `errors=coerce`, and errors are raised when `errors=raise`.

Because `downcast` is separate from the actual numeric conversion (you can only do it should you actually get numeric data at the end), I don't why it needs to be so harsh.  It's nice to have, and the main goal of converting to numeric (as the name indicates) has been achieved.

The other reason I don't want to be so harsh is because `com._maybe_downcast_to_dtype` itself is very lenient as the name suggests.  To make `downcast` harsh seems inconsistent with the design of that function, even though it is internal to `pandas`.

Casting to `float` is not that unreasonable if you consider `itemsize`, which is what underlies `com._maybe_downcast_to_dtype`:
~~~python
>>> import numpy as np
>>> np.dtype(np.float32).itemsize
4
>>> np.dtype(np.int64).itemsize
8
~~~